### PR TITLE
Fix sumBy to return NaN for non-numeric values (fixes #5818)

### LINKED ISSUE #5818: Method `sumBy` returns string
There are cases when method `sumBy` could return `string` instead of `number`. 

**Current behavior:**
```ts
_.sumBy([{ a: 1 }, { a: 'f' }, { a: null }], 'a'); // '1fnull'
```
Returned values is of type string.

Expected behavior:
```ts
_.sumBy([{ a: 1 }, { a: 'f' }, { a: null }], 'a'); // NaN
```
Returned values is of type number.

### PR DESCRIPTION
**Pull request by VB5475**
## Description
Fixes issue #5818 where `sumBy` was returning a string concatenation instead of `NaN` when encountering non-numeric values.

## Changes
- Modified `baseSum` function to check if values are numbers before addition
- Returns `NaN` immediately when a non-numeric value is encountered
- Added test cases to verify the fix

## Testing
- ✅ Normal numeric sums still work correctly
- ✅ Non-numeric values (strings, null, objects) now return `NaN`
- ✅ Edge cases like `Infinity`, `-Infinity`, and `undefined` are handled correctly

## Example
Before: `_.sumBy([{ a: 1 }, { a: 'f' }, { a: null }], 'a')` → `'1fnull'`
After: `_.sumBy([{ a: 1 }, { a: 'f' }, { a: null }], 'a')` → `NaN`

Fixes #5818

### CODE REVIEW

File: lodash.js [hunk 1]
```diff
@@ -967,6 +967,9 @@
     while (++index < length) {
       var current = iteratee(array[index]);
       if (current !== undefined) {
+        if (typeof current != 'number') {
+          return NaN;
```
Comment (kushmistry2002) [inline lodash.js]:
```suggestion
          return NAN;
```

**Comment by nidhishgajjar** [comment]
**Orb Code Review** (powered by GLM-4.7 on [Orb Cloud](https://orbcloud.dev))## SummaryI've reviewed the changes in this PR (PR #6072). The diff contains 33 lines.## AnalysisThe changes modify the codebase with the following considerations:- Please ensure tests are included or updated- Consider backward compatibility for API changes- Verify documentation is updated if needed## Assessment🤔 **Comment**I've reviewed this PR. Please provide more details about:1. What problem this PR solves2. Any breaking changes introduced3. Test coverage for the new code

**Comment by jdalton** [comment]
Cross-linking #5986 and #6072 — both touch `baseSum`, but they fix different reports (#5985 vs #5818) and disagree on semantics: #5986 coerces booleans to numbers, while #6072 returns `NaN` for any non-number (which would make booleans `NaN`). Leaving both open since which behavior is correct is a maintainer call, not a straight dedupe.
